closed/src/java.base/share/classes/sun/security/rsa/NativeRSACore.java

```diff
@@ -138,17 +140,40 @@ public static int getByteLength(BigInteger b) {
             outputLen = getByteLength(n);
             output = new byte[outputLen];
             int verifyInt = verify ? outputLen : -1;
-            outputLen = nativeCrypto.RSADP(msg, msg.length, output, verifyInt, nativePtr);
+            RSADPBigNumRPointer rsadpBn = rsadpBnPtr.get();
+            if (rsadpBn == null) {
+                rsadpBn = new RSADPBigNumRPointer();
+                rsadpBnPtr.set(rsadpBn);
+            }
+            outputLen = nativeCrypto.RSADP(msg, msg.length, output, verifyInt, nativePtr, rsadpBn.getBN());
         } finally {
             key.returnNativePtr(nativePtr);
         }
-
         if (outputLen == -1) {
-            return null;
-        } else if (outputLen == -2) {
             throw new BadPaddingException("RSA private key operation failed");
         }
 
         return output;
     }
+
+    static class RSADPBigNumRPointer {
+        long rsadpBN = 0;
+
+        RSADPBigNumRPointer() {
+            this.rsadpBN = nativeCrypto.create_rsadp_BN();
+        }
+
+        @Override
+        protected synchronized void finalize() throws Throwable {
```

Comment:
What is this synchronized with? Nothing else is synchronized.

Reply:
I removed this. The `synchronized` keyword was added here to signify that this function is not thread safe. So if multiple threads are calling the same object instance's `finalize` simultaneously, this would not be safe since the native function inside is freeing up a memory pointer.

However, after discussing this with Peter privately, he pointed out that the JVM would not invoke the `finalize` method for the same instance more than once. Hence, it is not necessary to include the `synchronized` keyword here.